src/lib/dev-environment/dev-environment-core.js

```diff
@@ -108,12 +100,41 @@ export async function createEnvironment( instanceData: NewInstanceData ) {
 		throw new Error( 'Environment already exists.' );
 	}
 
+	const preProcessedInstanceData = preProcessInstanceData( instanceData );
+
+	await prepareLandoEnv( preProcessedInstanceData, instancePath );
+}
+
+export async function updateEnvironment( instanceData: InstanceData ) {
+	const slug = instanceData.siteSlug;
+	debug( 'Will update an environment', slug, 'with instanceData: ', instanceData );
+
+	const instancePath = getEnvironmentPath( slug );
+
+	debug( 'Instance path for', slug, 'is:', instancePath );
+
+	const alreadyExists = fs.existsSync( instancePath );
+
+	if ( ! alreadyExists ) {
+		throw new Error( 'Environment doesn\'t exist.' );
+	}
+
+	const preProcessedInstanceData = preProcessInstanceData( instanceData );
+
+	await prepareLandoEnv( preProcessedInstanceData, instancePath );
+}
+
+function preProcessInstanceData( instanceData: InstanceData ): InstanceData {
+	const newInstanceData = {
```

Comment:
Does it have any side effects to modify the original object instead of having to create a new one?

Reply:
If you do:
```
const a = {};
const b = preprocess(a)
```
You don't expect a to be changed as well. [Pure functions ](https://en.wikipedia.org/wiki/Pure_function) are kinda easy to think about I feel like.

Reply:
Agreed, but copying an object over and over has an effect on performance. My question comes from the fact that if what we're doing here is taking an object and putting it through a series of functions that modify it (and those functions are only used here), it's fine to change the object that is passed by reference.

This is a more philosophical comment, so not worth blocking the PR because of it.